internal/service/emr/studio.go

```diff
@@ -0,0 +1,274 @@
+			},
+			"service_role": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"subnet_ids": {
+				Type:     schema.TypeSet,
+				MaxItems: 5,
+				MinItems: 1,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"tags":     tftags.TagsSchema(),
+			"tags_all": tftags.TagsSchemaComputed(),
+			"user_role": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"workspace_security_group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceStudioCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).EMRConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	input := &emr.CreateStudioInput{
+		AuthMode:                 aws.String(d.Get("auth_mode").(string)),
+		DefaultS3Location:        aws.String(d.Get("default_s3_location").(string)),
+		EngineSecurityGroupId:    aws.String(d.Get("engine_security_group_id").(string)),
+		Name:                     aws.String(d.Get("name").(string)),
+		ServiceRole:              aws.String(d.Get("service_role").(string)),
+		SubnetIds:                flex.ExpandStringSet(d.Get("subnet_ids").(*schema.Set)),
+		VpcId:                    aws.String(d.Get("vpc_id").(string)),
+		WorkspaceSecurityGroupId: aws.String(d.Get("workspace_security_group_id").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("idp_auth_url"); ok {
+		input.IdpAuthUrl = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("idp_relay_state_parameter_name"); ok {
+		input.IdpRelayStateParameterName = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("user_role"); ok {
+		input.UserRole = aws.String(v.(string))
+	}
+
+	if len(tags) > 0 {
+		input.Tags = Tags(tags.IgnoreAWS())
+	}
+
+	var result *emr.CreateStudioOutput
+	err := resource.Retry(tfiam.PropagationTimeout, func() *resource.RetryError {
+		var err error
+		result, err = conn.CreateStudio(input)
+		if tfawserr.ErrMessageContains(err, emr.ErrCodeInvalidRequestException, "entity does not have permissions to assume role") {
+			return resource.RetryableError(err)
+		}
+		if tfawserr.ErrMessageContains(err, emr.ErrCodeInvalidRequestException, "Service role does not have permission to access") {
+			return resource.RetryableError(err)
+		}
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if tfresource.TimedOut(err) {
+		result, err = conn.CreateStudio(input)
```

Comment:
Is there a reason you decided to not use CreateStudioWithContext in this new resource? This is just a question.

Reply:
nope, just inertia from previous PRs. need to start doing that.